config/v1/types_operatorhub.go

```diff
@@ -0,0 +1,84 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// OperatorHubKind is the string value of OperatorHub kind
+var OperatorHubKind = "OperatorHub"
+
+// OperatorHubSpec defines the desired state of OperatorHub
+// +k8s:openapi-gen=true
+type OperatorHubSpec struct {
+	// HubSources is the list of default OperatorSources and their configuration
+	HubSources []HubSource `json:"hubSources,omitempty"`
+}
+
+// OperatorHubStatus defines the observed state of OperatorHub
+// +k8s:openapi-gen=true
+type OperatorHubStatus struct {
+	// HubSourcesStatus encapsulates the result applying the configuration
+	HubSourcesStatus []HubSourceStatus `json:"hubSourcesStatus,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OperatorHub is the Schema for the operatorhubs API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +genclient:nonNamespaced
+type OperatorHub struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OperatorHubSpec   `json:"spec,omitempty"`
+	Status OperatorHubStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OperatorHubList contains a list of OperatorHub
+type OperatorHubList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OperatorHub `json:"items"`
+}
+
+// HubSource is used to specify the OperatorSource and its configuration
+type HubSource struct {
+	// Name is the name of one of the default OperatorSources
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// Disable is used to disable a default OperatorSource on cluster
+	Disable bool `json:"disable"`
+}
+
+// HubSourceStatus is used to reflect the current state of applying the
+// configuration to a default source
+type HubSourceStatus struct {
+	// Name is the name of one of the default OperatorSources
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+	// Configuration is the state of the default OperatorSources configuration
+	Configuration map[string]string `json:"configuration,omitempty"`
+	// Condition indicates success or failure in applying the configuration
+	Condition string `json:"condition,omitempty"`
```

Comment:
Wouldn't it be better to have a slice of conditions for each HubSource?

Reply:
`OperatorHubStatus` contains a slice of `HubSourceStatus` which maps to each `HubSource`

Reply:
+1 this doesn't follow the condition pattern of other resources.

Compare https://github.com/openshift/api/blob/f15120709e0ac8de84e11616d8f0cac54e8f52e3/config/v1/types_cluster_operator.go#L41.

We could rename this one to `Status`?

Reply:
I am a +1 on renaming `Condition` to `Status`. @shawn-hurley are you okay with that approach?

Reply:
+1 to renaming the condition field

conditions are supposed to be transient, abnormal-true states, and this hub status is showing the current state of all managed sources

Reply:
I am ok with the rename to `Status`